keyserver/listener.go

```diff
@@ -0,0 +1,127 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
```

Comment:
I think this would all be clearer as,

```
defer ln.Close()
for {
  switch {
    <-server.stop
    return
  }
  conn, err := ln.Accept()
  if err != nil {
    log.Printf("accept client: %s", err)
    continue
  }
  ...
}
```